src/common/snippets/src/pass/collapse_subgraph.cpp

```diff
@@ -192,9 +118,14 @@ auto has_supported_in_out(std::shared_ptr<Node> n) -> bool {
         if (in.get_partial_shape().is_static() && in.get_shape().size() > 6) {
```

Comment:
I don't think it should be argument of TokenizeSnippets. For me it looks like plugin specific condition and not all the plugins will need it. 
Is it possible to handle this limitation via TokenizeSnippets callback?

Reply:
First of all, my mistake - it has nothing to do with the number of available registers of course (but [tihs does](https://github.com/IvanNovoselov/openvino/blob/abb21f4aa47cc1a30e179d4c7bb8be22421bbe99/src/common/snippets/src/pass/collapse_subgraph.cpp#L474-L480)). It seems to me that this limitation is caused by scheduling implementation in MKLDNNSnippetNode, so its still plugin-specific.

Secondly, it is indeed possible to move this check (and also the [has_only_const_inputs](https://github.com/openvinotoolkit/openvino/pull/7936#discussion_r768008626)) to the plugin callback, so I did.

The last point is that we still have one plugin-specific hard-code in CreateSubgraph callback ([the mentioned num_reg limitation](https://github.com/IvanNovoselov/openvino/blob/abb21f4aa47cc1a30e179d4c7bb8be22421bbe99/src/common/snippets/src/pass/collapse_subgraph.cpp#L474-L480). It might be also possible to move it to callback, but refactoring of the transformation is required for this purpose, so I suggest to do it as a follow-up.